### PR TITLE
2024 Add sorting algo (insertion,radix,quick,heap,bucket) 

### PR DESCRIPTION
**Pull request by aditya-nv-06**
this pr contains sorting algorithms like insertion sort, radix sort, quick sort, heap sort and bucket sort in python .

### CODE REVIEW

**Review by FerdausPolok — COMMENTED** [review]
Hello @Aditya-NV-06!
Is this PR up for review? Could you please add a short description? Also your title says adding only about insertion_sort but I can see file chnages are beyond insertion_sort. Thanks.

**Review by FerdausPolok — CHANGES_REQUESTED** [review]
Please also add EOL (end of line) to all of the files. Thanks.

**Comment by aditya-nv-06** [comment]
can you elaborate the changes you want to make in this repo

**Comment by aditya-nv-06** [comment]
made the changes to the files

File: algorithm/sorting-algo-python/bucket_sort.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    return sorted_array
+
+# Example usage
+    arr = [0.897, 0.565, 0.656, 0.1234, 0.665, 0.3434]
+    print("Original array:", arr)
+    sorted_arr = bucket_sort(arr)
+    print("Sorted array:", sorted_arr)
+
+#end of the line
```
Comment (FerdausPolok) [inline algorithm/sorting-algo-python/bucket_sort.py]: By EOL we mostly refer to an empty new line at the bottom of the files.
If you do so, you no longer see the red circled minus (-) icon in GitHub file changes.
image

```suggestion
    sorted_arr = bucket_sort(arr)
    print("Sorted array:", sorted_arr)

```

File: algorithm/sorting-algo-python/heap_sort.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        heapify(arr, i, 0)
+
+
+
+# Driver code to test above
+    arr = [12, 11, 13, 5, 6, 7]
+    heap_sort(arr)
+    print("Sorted array is:", arr)
+
+#end of the line
```
Comment (FerdausPolok) [inline algorithm/sorting-algo-python/heap_sort.py]: Ditto.

FYI: By ditto we simple mean "Please do the same here as well"

File: algorithm/sorting-algo-python/insertion_sort.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+            arr[j + 1] = arr[j]
+            j -= 1
+        arr[j + 1] = key
+
+# Driver code to test above
+    arr = [12, 11, 13, 5, 6]
+    insertion_sort(arr)
+    print("Sorted array is:", arr)
+
+#end of the line
```
Comment (FerdausPolok) [inline algorithm/sorting-algo-python/insertion_sort.py]: ditto

File: algorithm/sorting-algo-python/quick_sort.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+
+
+# Driver code to test above
+    arr = [3, 6, 8, 10, 1, 2, 1]
+    print("Original array:", arr)
+    sorted_arr = quick_sort(arr)
+    print("Sorted array:", sorted_arr)
+
+#end of the line
```
Comment (FerdausPolok) [inline algorithm/sorting-algo-python/quick_sort.py]: ditto

File: algorithm/sorting-algo-python/radix_sort.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        exp *= 10
+
+
+
+# Driver code to test above
+    arr = [170, 45, 75, 90, 802, 24, 2, 66]
+    radix_sort(arr)
+    print("Sorted array is:", arr)
+
+#end of the line
```
Comment (FerdausPolok) [inline algorithm/sorting-algo-python/radix_sort.py]: ditto

**Review by FerdausPolok — CHANGES_REQUESTED** [review]

**Comment by aditya-nv-06** [comment]
changes are made in the files

**Comment by aditya-nv-06** [comment]
Correctly refactored the code this time as you said to me.

**Review by FerdausPolok — APPROVED** [review]
LGTM 🚀

**Pull request merged**